### PR TITLE
feat(useRetry): new function

### LINKED ISSUE #1649: feat(useRetry): retry something until it fulfills
### Clear and concise description of the problem

Currently I have an API call with `useFetch` and I need to refetch it in an interval until it is a 2xx response.
I came up with the idea of a new option `retry` for `useFetch` but I thought this could be advanced much more with a `useRetry` that wrappes `useFetch` but also something different.

### Suggested solution

Add a new utility composable `useRetry`

### Alternative

Only implement a `retry` option for `useFetch`

### Additional context

On discord I asked if this would be accepted and so I create this issue.

image

---

- [x] I'm willing to create the PR in next few hours/days, but I will also think about a useful concept first / collect some more experience with my current solution in my company's project.
  If I do not have the permission, please assign me to this GitHub issue.

---

This issue can be used to collect further ideas from other contributors.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by Shinigami92**
### Description

see #1649 

### Additional context

closes #1649 

---

### What is the purpose of this pull request? 

- [ ] Bug fix
- [x] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/core/useRetry/index.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+Retry something until fulfilled.
+
+## Usage
+
+```ts
+import { useFetch, useRetry } from '@vueuse/core'
+
+const { onFinish } = useRetry(
+  () => useFetch('https://example.com/api/users'),
```
Comment (antfu) [inline packages/core/useRetry/index.md:11]: I am not sure if that it's an ideal solution. If this function been called async, the composable will lost the context of current instance, which makes it failed to register cleanup functions and ends up with memory leak.

Comment (Shinigami92) [reply packages/core/useRetry/index.md:11]: So you have any hints or other VueUse examples were I can orient myself?

**Comment by Shinigami92** [comment]
@antfu @wheatjs please do not close this issue due to inactivity from my side. I'm in vacation mode for July and try to do at less as possible on GitHub for this month to rest from my first half year faker maintainance.

**Comment by Shinigami92** [comment]
dont have the time and prio to work anymore on that, sry :/

**Pull request closed without merging**